### ISSUE TITLE
Add examples from feature descriptions to proper ExampleTable

### ISSUE THREAD

**Issue by xrotwang** [grambank/grambank#58]
and possibly write feature descriptions in CLDF Markdown.

**Comment by HedvigS** [comment]
@johenglisch @xrotwang what would you like to do here next?

**Comment by johenglisch** [comment]
Hm, I suppose we could start by parsing the examples out of the feature descriptions and hoping that the formatting is uniform enough that we could do this automatically (incl. sources).

I'm not so sure CLDF markdown would work here – It makes a lot of sense for statically generated documents (books, webapps, etc.) but the wiki is treated very much like a mutable WYSIWYG-type thing:

 1. People reading the wiki expect the examples to just be there.  Any sort of link or ID will just feel clunky and noisy.
 2. People writing the wiki expect to be able to edit things left and right without worrying about a compilation step that they could break or that might overwrite their changes.

(btw: I think we talked about this already in some other issue buried in some git repo somewhere.  Can't find it though and I don't remember what the conclusion was… (<\_<)" )

**Comment by xrotwang** [comment]
@johenglisch valid points. There sure needs to be some trade off involved here. So

ad 1) Yes, I wouldn't go for CLDF markdown in the wiki pages (as they are in the wiki), but for the feature descriptions as they are packaged with the CLDF dataset.

ad 2) I would hope that editing of the wiki will slow down at some point - possibly that already happened. Otherwise we would have to do the parsing for each relase.

Overall, I still think machine-readable examples that are tied closely to something as well described as a Grambank feature would be a good resource. The fact that many of these examples are sourced from literature makes it even more so. It has this web-of-knowledge vibe :)

So yes, we should check how easy automated example extraction actually is, and take decisions once we know.

**Comment by johenglisch** [comment]
>I wouldn't go for CLDF markdown in the wiki pages (as they are in the wiki),
>but for the feature descriptions as they are packaged with the CLDF dataset

So, basically add CLDF markdown to the *Description* column of the *ParameterTable*?  (Which would also mean that we'd have to make the webapp ‘CLDF markdown aware’)

>we would have to do the parsing for each release

I just kinda assumed that's how things would end up anyway -- I'd just do the parsing during *makecldf* (and if that fails, we tell someone to fix the wiki – or roll back to a known working version).

>we should check how easy automated example extraction actually is

Yeah, the few wiki pages I looked at seem formulaic enough that we might be in luck.

>and take decisions once we know

Yup, I'll see if I can make a functioning example table first and then we can see if we can make some CLDF markdown out of it.

**Comment by HedvigS** [comment]
I don't know if it matters, but I don't know the motivation here. I think I can be more helpful here if I know the motivation and intended user.

**Comment by johenglisch** [comment]
To me it's kind of a *might as well* thing:

 * We have glossed examples in the feature descriptions, so we might as well add them to the data set and link them properly to their language and sources.
 * Once we have the examples in our data set we might as well refer to them by ID in the feature description rather than keeping the copy-paste version around.
 * As a bonus, examples retrieved from the data set can be rendered properly instead of dumping them into those janky source code blocks.

By the by, I remembered where the example table came up before and found the issue: https://github.com/clld/grambank/issues/80

**Comment by xrotwang** [comment]
Yes, the motivation is extracting well-described structured data. There have been projects before, trying to aggregate IGT data (e.g. ODIN). And that data has been used, e.g. in a shared task for gloss prediction, etc.

**Comment by HedvigS** [comment]
> By the by, I remembered where the example table came up before and found the issue: [clld/grambank#80](https://github.com/clld/grambank/issues/80)
Aaah, it's a thing in relation to that! That makes sense, now I get it.

> Yes, the motivation is extracting well-described structured data. There have been projects before, trying to aggregate IGT data (e.g. ODIN). And that data has been used, e.g. in a shared task for gloss prediction, etc.
Aha okay, I didn't know abut that.

**Comment by HedvigS** [comment]
In relation other 

> ad 2) I would hope that editing of the wiki will slow down at some point - possibly that already happened. Otherwise we would have to do the parsing for each relase.

Wiki editing has def slowed down. It is still necessary however to continue curating the wiki articles as we've done, it's what works well for patrons and coders.

**Comment by HedvigS** [comment]
@xrotwang @johenglisch  what would you like to do here next?

**Comment by johenglisch** [comment]
I've been working on this.  I have a local branch with a draft of an example extractor (took longer than expected – the formatting isn't nearly as consistent as it seems at a first glance).  I'll have something soon™.  Also, I'll probably go in and fix a few typos in the wiki that I found in the process).

**Comment by johenglisch** [comment]
Closed in <https://github.com/grambank/pygrambank/pull/121>
